ansible/roles/vmware-template/tasks/create.yml

```diff
@@ -1,80 +1,127 @@
 ---
 
-- name: Define templates' name and annotation
+- name: Define templates' variables
   set_fact:
     template_name: "Template-{{ vm_role }}-{{ '%Y%m%d%H%M' | strftime(ansible_date_time.epoch) }}"
     template_annotation: "Template for {{ vm_role }} created on {{ '%Y-%m-%d %H:%M:%S' | strftime(ansible_date_time.epoch) }}"
+    template_folder: "{{ vcenter_datacenter }}/vm/{{ vmware_folder }}"
+    template_source: "{{ vm_template }}"
 
-- name: Create Template VM for {{ vm_role }}
-  register: new_vm
+- name: Create template VM from base template
   vmware_guest:
     hostname: "{{ vcenter_hostname }}"
-    username: "{{ vcenter_user }}"
+    username: "{{ vcenter_username }}"
     password: "{{ vcenter_password }}"
     validate_certs: "{{ vcenter_validate_certs }}"
-
     datacenter: "{{ vcenter_datacenter }}"
     cluster: "{{ vcenter_cluster }}"
-    folder: "{{ vmware_folder }}"
 
+    folder: "{{ template_folder }}"
     name: "{{ template_name }}"
+    template: "{{ template_source }}"
     annotation: "{{ template_annotation }}"
-    template: "{{ vm_template }}"
     hardware: "{{ vm_hardware }}"
     state: poweredon
     customization:
       hostname: "tmpl-{{ vm_role }}"
       password: "Contrail123!"
-
     networks: "{{ vm_networks }}"
 
-    wait_for_ip_address: yes
+- name: Wait for VMware tools to come up after cloning base template
+  vmware_guest_tools_wait:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: "{{ vcenter_validate_certs }}"
+    folder: "{{ template_folder }}"
+    name: "{{ template_name }}"
+
+# NOTE: "Workaround". One way to check if customization has finished is to check hostname through
```

Comment:
What about customization during the deployment of template? Shouldn't we also wait for customization?

Reply:
Playbook responsible for template deployment does not do any reboot shenanigans, which are done in create template playbook. After deploying a VM from template it waits for IP of the machine, so it should work just fine.